From 1c4a370d31c39572e3a7e7ef1422a3e2badca924 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek.chauhan@gmail.com>
Date: Fri, 14 Jan 2011 07:40:51 +0530
Subject: [PATCH 2/2] Port to latest gtk+:3

* gtk_combo_box_append_text(), gtk_combo_box_get_active_text() were removed
* gtk_quit_add() and GtkFunction were removed
  - gtk_quit_add() shouldn't be needed since uninit is called atexit()
* Require gtk+-2.91.1 or gtk+-2.23.0 for this
---
 agent/seahorse-agent-main.c            |    2 --
 configure.in                           |    4 ++--
 plugins/nautilus/seahorse-tool-files.c |    7 +++----
 3 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/agent/seahorse-agent-main.c b/agent/seahorse-agent-main.c
index 5f7c95e..78e24d8 100644
--- a/agent/seahorse-agent-main.c
+++ b/agent/seahorse-agent-main.c
@@ -239,8 +239,6 @@ int main(int argc, char* argv[])
 
     gtk_init_with_args (&argc, &argv, _("Encryption Key Agent (Seahorse)"), (GOptionEntry *) options, GETTEXT_PACKAGE, NULL);
 
-    gtk_quit_add (0, (GtkFunction) seahorse_agent_uninit, NULL);
-    
     seahorse_agent_prefork ();
 
     if (seahorse_agent_execvars && 
diff --git a/configure.in b/configure.in
index 42ef37b..a55f6d0 100644
--- a/configure.in
+++ b/configure.in
@@ -44,11 +44,11 @@ AC_MSG_RESULT([$with_gtk])
 case "$with_gtk" in
 	2.0)
 		GTK_API_VERSION=2.0
-		GTK_REQUIRED=2.18.0
+		GTK_REQUIRED=2.23.0
 	;;
 	3.0)
 		GTK_API_VERSION=3.0
-		GTK_REQUIRED=2.90.0
+		GTK_REQUIRED=2.91.1
 	;;
 esac
 
diff --git a/plugins/nautilus/seahorse-tool-files.c b/plugins/nautilus/seahorse-tool-files.c
index b0f6045..0dba4d7 100644
--- a/plugins/nautilus/seahorse-tool-files.c
+++ b/plugins/nautilus/seahorse-tool-files.c
@@ -432,8 +432,8 @@ prepare_dialog (FilesCtx *ctx, guint nfolders, guint nfiles, GFileInfo *info, gc
         save_type_list = save_type;
         
         for (i = 0; save_type_list[i] != FR_FILE_TYPE_NULL; i++) {
-    		gtk_combo_box_append_text (GTK_COMBO_BOX (combo),
-    					   file_type_desc[save_type_list[i]].ext);
+    		gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combo),
+    					        file_type_desc[save_type_list[i]].ext);
 		    if (strcmp(ext, file_type_desc[save_type_list[i]].ext) == 0)
 		        gtk_combo_box_set_active (GTK_COMBO_BOX (combo), i);
     	}
@@ -473,8 +473,7 @@ get_results (SeahorseWidget *swidget)
         name = gtk_entry_get_text (GTK_ENTRY (w));
         
         w = GTK_WIDGET (seahorse_widget_get_widget (swidget, "package-extension"));
-        ext = gtk_combo_box_get_active_text (GTK_COMBO_BOX (w));
-        
+        ext = gtk_combo_box_text_get_active_text (GTK_COMBO_BOX_TEXT (w));
         
         /* No paths */
         t = strrchr(name, '/');
-- 
1.7.3.4

